### PR TITLE
[Android] implement TCP pairing handshake for issue #96

### LINKED ISSUE #96: [Android] TCP: Pairing Handshake Integration
**Parent Issue:** #60

**Description:**
Implement the TCP portion of the pairing handshake as specified in `Pairing Process.md` Section 2. This integrates with `PairingManager` to complete Phase 2 and Phase 3 of device pairing.

**Related Requirements:** NET1, SYS

**Pairing Flow (from Pairing Process.md Section 2):**
1. **Phase 1:** UDP Discovery (handled by #59)
2. **Phase 2(b):** Android establishes TCP connection and sends `PAIRING_REQUEST` (0x20)
3. **Phase 3(a):** Server responds with `PAIRING_ACK` (0x21)
4. **Phase 3(b):** Android proceeds to HTTP registration

**Tasks:**
- [ ] Add pairing handshake to `TcpSocketManager`
  - Send `PAIRING_REQUEST` immediately after TCP connection established
  - Include device ID (from device settings) as UTF-8 operand
  - Wait for `PAIRING_ACK` response
- [ ] Create `PairingCallback` interface
  ```java
  public interface PairingCallback {
      void onPairingSuccess();
      void onPairingFailed(String reason);
      void onPairingTimeout();
  }
  ```
- [ ] Implement pairing timeout
  - Configurable timeout (default: 10 seconds)
  - Transition to failed state if no ACK received
- [ ] Integrate with `PairingManager` (issue 6.10 equivalent)
  - Notify on successful pairing (trigger Phase 3b - HTTP registration)
  - Notify on failure (allow retry or show error)
- [ ] Handle pairing edge cases
  - Connection lost during handshake
  - Invalid response from server
  - Multiple pairing attempts
- [ ] Add pairing state to `ConnectionState` or create separate `PairingState`
  - `NOT_PAIRED`, `PAIRING_IN_PROGRESS`, `PAIRED`, `PAIRING_FAILED`
- [ ] Write unit tests
  - Successful pairing flow
  - Timeout handling
  - Connection loss during pairing
  - Invalid response handling

**Acceptance Criteria:**
- [ ] `PAIRING_REQUEST` sent on TCP connection
- [ ] `PAIRING_ACK` received and processed
- [ ] `PairingCallback` notifies success/failure
- [ ] Pairing timeout handled
- [ ] Integration point for PairingManager
- [ ] Edge cases handled gracefully
- [ ] Pairing state exposed
- [ ] Unit tests for pairing flow
- [ ] Checkstyle compliant
- [ ] Javadoc for all public methods

**Technical Notes:**
- Device ID: Use `Settings.Secure.ANDROID_ID` or generate UUID on first launch
- Pairing request: `new PairingRequestMessage(deviceId)`
- Timeout: Use `CountDownLatch` with `await(timeout, TimeUnit.SECONDS)` or callback-based
- After pairing success, heartbeat should start automatically

**Dependencies:**
- #91 - Pairing message classes
- #93 - TcpSocketManager for connection
- #94 - Listener system to receive PAIRING_ACK
- #59 - UDP Discovery for server IP/port

### PR DESCRIPTION
**Pull request by PriyaBargota**
## Summary
- Implement `PairingState` enum to track pairing lifecycle (NOT_PAIRED, PAIRING_IN_PROGRESS, PAIRED, PAIRING_FAILED, PAIRING_TIMEOUT)
- Add `PairingCallback` interface for pairing event notifications
- Create `PairingConfig` class for configurable timeout and retry settings
- Implement `PairingManager` to orchestrate TCP pairing handshake as specified in Pairing Process.md Section 2

## Test plan
- [x] Unit tests for `PairingState` enum values
- [x] Unit tests for `PairingConfig` value clamping and builder methods
- [x] Unit tests for `PairingManager` pairing flow, callbacks, and error handling
- [x] All tests pass locally
- [x] Checkstyle passes

Closes #96

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: android/app/src/main/java/com/manuscripta/student/network/tcp/PairingManager.java [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Manages the TCP pairing handshake with the teacher server.
+ *
+ * <p>The pairing flow as specified in {@code Pairing Process.md} Section 2:
```
[the remark this answers is not in the document]
Comment (PriyaBargota) [reply android/app/src/main/java/com/manuscripta/student/network/tcp/PairingManager.java]: Minor style suggestion - keeping as-is since the @code markup makes it clear this references documentation.

File: android/app/src/main/java/com/manuscripta/student/network/tcp/PairingManager.java [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    private void startTimeoutTimer() {
+        cancelTimeoutTimer();
+
+        scheduler = Executors.newSingleThreadScheduledExecutor();
+        timeoutFuture = scheduler.schedule(
+                this::handleTimeout,
+                config.getTimeoutMs(),
+                TimeUnit.MILLISECONDS
+        );
+    }
```
[the remark this answers is not in the document]
Comment (PriyaBargota) [reply android/app/src/main/java/com/manuscripta/student/network/tcp/PairingManager.java]: The scheduler is shut down in cancelTimeoutTimer() which is called before creating a new one in startTimeoutTimer(). Added explicit cancelTimeoutTimer() call in retry path to ensure cleanup. Creating a persistent scheduler would require more lifecycle management with destroy() - current approach is simpler.

**Review by WillStephenn — APPROVED** [review]
